unit-tests/live/frames/test-fps-permutations.py

```diff
@@ -0,0 +1,304 @@
+# License: Apache 2.0. See LICENSE file in root directory.
+# Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+# test:donotrun:!nightly
+# test:timeout 750
+# timeout - on the worst case, we're testing on D585S, which have 8 sensors, so:
+# timeout = (8 choose 2) * 24 + 24 = 696
+# 8 choose 2 tests to do (one for each pair), plus one for all sensors on, each test takes 24 secs
+
+from rspy import test, log
+import time
+import itertools
+import math
+
+
+# To add a new test mode (ex. all singles or triplets) only add it below and on should_test
+# add it in get_time_est_string for an initial time calculation
+# run perform_fps_test(sensor_profiles_array, [mode])
+ALL_PERMUTATIONS = 1
+ALL_PAIRS = 2
+ALL_SENSORS = 3
+
+# global variable used to count on all the sensors simultaneously
+count_frames = False
+
+# tests parameters
+TEST_ALL_COMBINATIONS = False
+seconds_till_steady_state = 4
+seconds_to_count_frames = 20
+
+
+##########################################
+# ---------- Helper Functions ---------- #
+##########################################
+def get_resolution(profile):
+    return profile.as_video_stream_profile().width(), profile.as_video_stream_profile().height()
+
+
+def check_fps_pair(fps, expected_fps):
+    delta_Hz = expected_fps * 0.05  # Validation KPI is 5%
+    return (fps <= (expected_fps + delta_Hz) and fps >= (expected_fps - delta_Hz))
+
+
+def get_expected_fps(sensor_profiles_dict):
+    """
+    For every sensor, find the expected fps according to its profiles
+    Return a dictionary between the sensor and its expected fps
+    """
+    expected_fps_dict = {}
+    # print(sensor_profiles_dict)
+    for key in sensor_profiles_dict:
+        avg = 0
+        for profile in sensor_profiles_dict[key]:
+            avg += profile.fps()
+
+        avg /= len(sensor_profiles_dict[key])
+        expected_fps_dict[key] = avg
```

Comment:
What is this avarage?

Reply:
Can we use default profiles instead?

Reply:
We can't use default profiles because some streams might be missed (on my D435 & D455 Stereo Module, only the depth is default, so IR is not tested), and besides, all streams on the same sensor must have the same resolution and I'm not sure this is always the case if we take the default.
The average is for when we can't find a mutual resolution and fps for all streams on a sensor. In that case we are looking for mutual resolution (without fps), so here we calculate what is the expected fps in that case.
I do think that most of the time we will find a mutual resolution and fps, so the average will be equal to that fps anyway :)